### PR TITLE
feat(metris): Record a number of metrics

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Check duration
 - Checks per bucket
 - Check status
 - Check failure reason

Part of https://github.com/getsentry/uptime-checker/issues/98

### CODE REVIEW

File: src/scheduler.rs [hunk 1]
```diff
@@ -30,6 +32,54 @@ pub fn run_scheduler(
...
+    } = result;
+
+    // Record duration of check
+    if let Some(duration) = duration {
+        Metric::timing("http_checker.duration", duration.to_std().unwrap()).send();
+    }
+
+    // Record time between scheduled and actual check
+    Metric::timing(
+        "http_checker.duration",
```
Comment (wedamija) [inline src/scheduler.rs:53]: Need to change this name? `drift` or something?

File: src/scheduler.rs [hunk 2]
```diff
@@ -30,6 +32,54 @@ pub fn run_scheduler(
...
+        scheduled_check_time,
+        actual_check_time,
+        duration,
+        status_reason,
+        ..
+    } = result;
+
+    // Record duration of check
+    if let Some(duration) = duration {
+        Metric::timing("http_checker.duration", duration.to_std().unwrap()).send();
```
Comment (wedamija) [inline src/scheduler.rs:48]: Should we have tags with the result on these (and all metrics)?

**Review by wedamija — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
nvm we're not using this, we want datadog

File: src/scheduler.rs (hunk 2 above)
Comment (evanpurkhiser) [reply src/scheduler.rs:48]: good idea 👍

File: src/scheduler.rs (hunk 1 above)
Comment (evanpurkhiser) [reply src/scheduler.rs]: Changed to `delay`

File: src/scheduler.rs [hunk 3]
```diff
@@ -30,6 +31,63 @@ pub fn run_scheduler(
...
+        "check_result.status",
+        "status" => status_label,
+        "failure_reason" => failure_reason.unwrap_or("ok"),
+    )
+    .increment(1);
+
+    // Record failure reason
+    if let Some(failure_reason) = failure_reason {
+        metrics::counter!("check_result.failure_reason", "reason" => failure_reason).increment(1);
+    }
```
Comment (wedamija) [inline src/scheduler.rs:88]: We don't really need this do we? Isn't it inherently included in the `status` counter?

File: src/scheduler.rs [hunk 4]
```diff
@@ -30,6 +31,63 @@ pub fn run_scheduler(
...
+        "check_result.delay_ms",
+        "histogram" => "timer",
+        "status" => status_label,
+        "failure_reason" => failure_reason.unwrap_or("ok"),
+    )
+    .record((*actual_check_time - *scheduled_check_time).num_milliseconds() as f64);
+
+    // Record status of the check
+    metrics::counter!(
+        "check_result.status",
```
Comment (wedamija) [inline src/scheduler.rs:79]: I think maybe this should be `check_result.processed` or something? The status is already a tag

File: src/scheduler.rs (hunk 3 above)
Comment (evanpurkhiser) [reply src/scheduler.rs:88]: hmm, this one only logs when there's a real failure. I guess we should be able to exclude the `none` and have a failure only chart in dd?

**Pull request merged**